continuity.api/src/main/java/org/continuity/api/entities/config/TaskDescription.java

```diff
@@ -23,6 +23,9 @@
 	@JsonProperty("long-term-use")
 	private boolean longTermUse;
 
+	@JsonProperty("modularization-options")
+	private ModularizationOptions modularizationOptions;
```

Comment:
If I get it right, you simply forward the modularization options to the services and every service decides on its own whether it should modularize something. Did you think about letting the orchestrator decide which services should apply modularization and telling the services explicitly? Then, the orchestrator had the control.